modules/REST/V1/Languages.php

```diff
@@ -0,0 +1,803 @@
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}/(?P<slug>[a-z_-]+)",
+			array(
+				'args'   => array(
+					'slug'    => array(
+						'description' => __( 'Language code - preferably 2-letters ISO 639-1 (for example: en).', 'polylang' ),
+						'type'        => 'string',
+					),
+				),
+				$readable,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items( $request ) {
+		$languages = array();
+
+		foreach ( $this->languages->get_list() as $language ) {
+			$languages[] = $this->prepare_item_for_response( $language, $request );
+		}
+
+		return rest_ensure_response( $languages );
+	}
+
+	/**
+	 * Creates one language from the collection.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function create_item( $request ) {
+		if ( isset( $request['term_id'] ) ) {
+			return new WP_Error(
+				'rest_exists',
+				__( 'Cannot create existing language.', 'polylang' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		// At this point, `$request['locale']` is provided (but maybe not valid).
+		$prepared = $this->prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared ) ) {
+			return $prepared;
+		}
+
+		/**
+		 * @phpstan-var object{
+		 *    locale: non-empty-string,
+		 *    slug: non-empty-string,
+		 *    name: non-empty-string,
+		 *    rtl: bool,
+		 *    term_group: int,
+		 *    flag: non-empty-string,
+		 *    no_default_cat: bool
+		 * } $prepared
+		 * @phpstan-var array{
+		 *    locale: non-empty-string,
+		 *    slug: non-empty-string,
+		 *    name: non-empty-string,
+		 *    rtl: bool,
+		 *    term_group: int,
+		 *    flag: non-empty-string,
+		 *    no_default_cat: bool
+		 * } $args
```

Comment:
Is it possible to create a type of https://github.com/polylang/polylang/blob/ce8452c0dd20c64551d2be333f693c3fb6b4df17/include/Model/Languages.php#L144-L152, import it and use it here?

Reply:
Possible but that's forcing the type `add()` wants, instead of providing the type `prepare_item_for_database()` returns (they're not exactly the same).